### PR TITLE
Updates SDCard capsule to Tock 2.0 system call API

### PR DESCRIPTION
**Pull request by phil-levis**
### Pull Request Overview

This pull request updates the SDCard capsule to the 2.0 system call API.

### Testing Strategy

This pull request has not been tested. No boards have SDCards on them. It is one of the capsules that will be tested for the merge into master.

### TODO or Help Wanted

Testing.

### Documentation Updated

- [x] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [x] Ran `make prepush`.

### CODE REVIEW

**Review by hudson-ayers — APPROVED** [review]

File: capsules/src/sdcard.rs [hunk 1]
```diff
@@ -1441,7 +1443,7 @@ impl<'a, A: hil::time::Alarm<'a>> SDCardClient for SDCardDriver<'a, A> {
         self.app.map(|app| {
             let mut read_len = 0;
             self.kernel_buf.map(|data| {
-                app.read_buffer.as_mut().map(|read_buffer| {
+                app.read_buffer.mut_map_or(0, |read_buffer| {
```
Comment (brghena) [inline capsules/src/sdcard.rs:1446]: I agree that this code is functionally identical to the previous code.

File: capsules/src/sdcard.rs [hunk 2]
```diff
@@ -1451,127 +1453,138 @@ impl<'a, A: hil::time::Alarm<'a>> SDCardClient for SDCardDriver<'a, A> {
...
             }
+            _ => Err((slice, ErrorCode::NOSUPPORT)),
+        }
+    }
 
-            // Pass write buffer in from application
-            1 => {
-                self.app.map(|app| app.write_buffer = slice);
-                ReturnCode::SUCCESS
+    fn allow_readonly(
```
Comment (brghena) [inline capsules/src/sdcard.rs:1502]: This is a good change that should be functionally identical anyways.

**Review by brghena — APPROVED** [review]
This looks good. I will test it as soon as I can.

**Pull request merged**